opensearch_upload/workflow_metrics_uploader.py

```
@@ -0,0 +1,795 @@
+                jobs_data = self.fetch_job_details(run_id)
+                if jobs_data and "jobs" in jobs_data:
+                    for job in jobs_data["jobs"]:
+                        self.upload_job_metrics(job, attempt_data)
+                        
+                        # Upload steps for this job
+                        for step_index, step in enumerate(job.get("steps", [])):
+                            self.upload_step_metrics(step, job, attempt_data, step_index)
+                
+                sleep(0.5)  # Rate limiting
+                
+            except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)